### PR TITLE
feat: sentry devservices

### PR DESCRIPTION
**Pull request by mattrobenolt**

### CODE REVIEW

File: src/sentry/conf/server.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def env(key, default='', type=None):
         'NAME': 'sentry',
         'USER': 'postgres',
         'PASSWORD': '',
-        'HOST': '',
+        'HOST': '127.0.0.1',
```
Comment (mattrobenolt) [inline src/sentry/conf/server.py]: Changing the default value here to not use a unix socket, and this should be backwards compatible in every way.

File: src/sentry/runner/commands/devservices.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    if not exclude:
+        exclude = set()
+    if 'kafka' in settings.SENTRY_EVENTSTREAM:
+        pass
+    elif 'snuba' in settings.SENTRY_EVENTSTREAM:
+        click.secho('! Skipping kafka and zookeeper since your eventstream backend does not require it', err=True, fg='cyan')
+        exclude |= {'kafka', 'zookeeper'}
+    else:
+        click.secho('! Skipping kafka, zookeeper, snuba, and clickhouse since your eventstream backend does not require it', err=True, fg='cyan')
+        exclude |= {'kafka', 'zookeeper', 'snuba', 'clickhouse'}
```
Comment (tkaemming) [inline src/sentry/runner/commands/devservices.py]: lil bit of duplication here

**Review by tkaemming — APPROVED** [review]
:nice:

Some minor nits inline but this is going to be really helpful

File: src/sentry/runner/commands/devservices.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    """
+    Manage dependent development services required for Sentry.
+
+    Do not use in production!
+    """
+
+
+@devservices.command()
+@click.option('--project', default='sentry')
+@click.option('--exclude', type=Set, help='Services to ignore and not run.')
```
[the remark this answers is not in the document]
Comment (mattrobenolt) [reply src/sentry/runner/commands/devservices.py]: We do similar in other CLI args. I just find this more concise vs doing `--exclude=foo --exclude=bar`. Specifically we do this in `sentry run worker` to exclude queues, etc.

[the remark this answers is not in the document]
Comment (tkaemming) [reply src/sentry/runner/commands/devservices.py]: Fair enough

**Comment by tkaemming** [comment]
:toot:

**Pull request merged**